### PR TITLE
fix(core): fix structure type attributes

### PR DESCRIPTION
**Pull request by patricksmms**
This PR is to sort out the attributes functionalities.

This was implemented before, but was broken https://github.com/apiaryio/api-elements.js/pull/561/files#diff-ed3f4907fe76fdf2e90227edd1bba42cL70

### CODE REVIEW

File: packages/api-elements/lib/define-value-of.js [hunk 1]
```diff
@@ -24,20 +25,35 @@ const {
...
  */
-function mapValue(e, f, elements) {
+function mapValue(e, f, inheritedAttrs, elements) {
   if (e === undefined) {
     return undefined;
   }
 
+  const attrs = {
+    isFixed: inheritedAttrs.isFixed || isFixed(e),
+    isFixedType: (inheritedAttrs.isFixed && isPrimitive(e)) || isFixedType(e),
```
Comment (kylef) [inline packages/api-elements/lib/define-value-of.js:51]: Fixed-type isn't an inherited type attribute, I think this part should be removed?

**Review by kylef — CHANGES_REQUESTED** [review]

File: packages/api-elements/lib/define-value-of.js (hunk 1 above)
Comment (patricksmms) [reply packages/api-elements/lib/define-value-of.js]: Indeed. Addressed in ef0df9c.

I'm now realizing it's missing tests for [required](https://github.com/apiaryio/api-elements.js/pull/564/files#diff-ed3f4907fe76fdf2e90227edd1bba42cR158).

**Review by kylef — APPROVED** [review]

File: packages/api-elements/lib/utils.js [hunk 2]
```diff
@@ -32,11 +32,18 @@ function hasTypeAttribute(e, attribute) {
 }
 
 /**
- * Check if element has 'fixed' or 'fixedType' typeAttribute set
+ * Check if element has 'fixed' typeAttribute set
  * @param {element} e - element
  * @return {boolean}
  */
-const isFixed = e => hasTypeAttribute(e, 'fixed') || hasTypeAttribute(e, 'fixedType');
+const isFixed = e => hasTypeAttribute(e, 'fixed');
```
Comment (kylef) [inline packages/api-elements/lib/utils.js:39]: Unrelated to PR but just something I noticed while reading this again, would be nice to make `hasTypeAttribute` curried and in opposite argument order. I can make a PR once these are merged

```
const isFixed = hasTypeAttribute('fixed');
```

**Review by kylef — APPROVED** [review]
Looks good to me!

File: packages/api-elements/lib/utils.js (hunk 2 above)
Comment (patricksmms) [reply packages/api-elements/lib/utils.js:39]: I can do that on the [other PR](https://github.com/apiaryio/api-elements.js/tree/patricksmms/improve-define-value-of)

**Pull request merged**